balancer/weightedroundrobin/weightedroundrobin.go

```diff
@@ -19,11 +19,42 @@
 // Package weightedroundrobin defines a weighted roundrobin balancer.
 package weightedroundrobin
 
+import (
+	"google.golang.org/grpc/resolver"
+)
+
 // Name is the name of weighted_round_robin balancer.
 const Name = "weighted_round_robin"
 
-// AddrInfo will be stored inside Address metadata in order to use weighted roundrobin
-// balancer.
+// attributeKey is the type used as the key to store AddrInfo in the Attributes
+// field of resolver.Address.
+type attributeKey struct{}
+
+// AddrInfo will be stored inside Address metadata in order to use weighted
+// roundrobin balancer.
 type AddrInfo struct {
 	Weight uint32
 }
+
+// SetAddrInfo sets addInfo in the Attributes field of addr.
+// This is an EXPERIMENTAL API.
+func SetAddrInfo(addrInfo AddrInfo, addr *resolver.Address) {
```

Comment:
Please update this to return a `resolver.Address` and not accept a pointer.

Reply:
Done.